httpx/utils.py

```diff
@@ -256,6 +260,23 @@ def unquote(value: str) -> str:
     return value[1:-1] if value[0] == value[-1] == '"' else value
 
 
+K = typing.TypeVar("K")
+V = typing.TypeVar("V")
+
+
+def flatten_on_values(
+    items: typing.Iterable[
+        typing.Union[typing.Tuple[K, typing.Union[V, typing.Sequence[V]]]]
+    ]
```

Comment:
I'd probably prefer for this not to be genericly typed, and to instead explicitly be `flatten_queryparam_values`.
Right now we're only using it in one place, and it'd be clearly to me what it does if it's more explcitly focused.

I'd also rather we switch the function around slightly so that it returns a list, rather than a generator. (Marginally less elgant in the implementation, but slightly more helpful at the point-of-use)

Reply:
A brief docstring too would make sense. As with `str_query_param` I'd *probably* prefer not to have the indirection to `PrimitiveData`, but I'm not absolute on that one.

Reply:
Agreed that specializing this helper more is a neat idea.

Reply:
Wouldn't mind a docstring on this, just "Given queryparams as a mapping, return a flat list of two-tuples representing each item, eg... *insert simple example case*"

Otherwise all look *ace* here!

Reply:
Docstring's in!